pkg/secrethub/client.go

```diff
@@ -290,3 +291,34 @@ func (c *Client) userAgent() string {
 
 	return userAgent
 }
+
+// Resolve fetches the value of a secret, when the `ref` parameter has the
+// format `secrethub://<path>`. Otherwise it returns `ref` unchanged, as an array of bytes.
+func (c *Client) Resolve(ref string) ([]byte, error) {
+	bits := strings.Split(ref, "://")
+	if len(bits) == 2 && bits[0] == "secrethub" {
+		secret, err := c.Secrets().Read(bits[1])
+		if err != nil {
+			return []byte{}, err
+		}
+		return secret.Data, nil
+	}
+	return []byte(ref), nil
+}
+
+// ResolveEnv takes a map of environment variables and replaces the values of those
+// which store references of secrets in SecretHub (`secrethub://<path>`) with the value
+// of the respective secret. The other entries in the map remain untouched.
+func (c *Client) ResolveEnv() (map[string]string, error) {
+	envVars := os.Environ()
```

Comment:
Should the user be in control of what's provided as input here or should it always get the environment?

Use case I'm thinking of is when you have other functions that manipulate the environment, you may want to pass that result in or something like that.

Something along the lines of secrethub.ResolveEnv(os.Environ()). Added benefit is less magic, downside is more typing.

But to be clear, I'm not at all certain about this design decision. It's just an idea.

Reply:
I adjusted the function to accept a parameter as you mentioned.